File: app-android/app/src/main/java/com/health2u/data/mapper/HealthInsightMapper.kt
package com.health2u.data.mapper

import com.health2u.data.local.entity.HealthInsightEntity
import com.health2u.data.remote.dto.HealthInsightDto
import com.health2u.domain.model.HealthInsight

// Entity to Domain
fun HealthInsightEntity.toDomain(): HealthInsight = HealthInsight(
    id = id,
    userId = userId,
    type = type,
    title = title,
    description = description,
    metricValue = metricValue,
    timestamp = timestamp,
    createdAt = createdAt
)

// Domain to Entity
fun HealthInsight.toEntity(): HealthInsightEntity = HealthInsightEntity(
    id = id,
    userId = userId,
    type = type,
    title = title,
    description = description,
    metricValue = metricValue,
    timestamp = timestamp,
    createdAt = createdAt
)

// DTO to Domain
fun HealthInsightDto.toDomain(): HealthInsight = HealthInsight(
    id = id,
    userId = userId,
    type = type,
    title = title,
    description = description,
    metricValue = metricValue,
    timestamp = timestamp,
    createdAt = createdAt
)

// Domain to DTO
fun HealthInsight.toDto(): HealthInsightDto = HealthInsightDto(
    id = id,
    userId = userId,
    type = type,
    title = title,
    description = description,
    metricValue = metricValue,
    timestamp = timestamp,
    createdAt = createdAt
)

File: app-ios/Sources/Health2u/Data/Mapper/HealthInsightMapper.swift
import Foundation

extension HealthInsightDTO {
    public func toDomain() -> HealthInsight {
        HealthInsight(
            id: id,
            userId: userId,
            type: type,
            title: title,
            description: description,
            metricValue: metricValue,
            timestamp: timestamp,
            createdAt: createdAt
        )
    }
}

extension HealthInsight {
    public func toDTO() -> HealthInsightDTO {
        HealthInsightDTO(
            id: id,
            userId: userId,
            type: type,
            title: title,
            description: description,
            metricValue: metricValue,
            timestamp: timestamp,
            createdAt: createdAt
        )
    }

    public func toEntity() -> HealthInsightEntity {
        HealthInsightEntity(
            id: id,
            userId: userId,
            type: type,
            title: title,
            descriptionText: description,
            metricValue: metricValue,
            timestamp: timestamp,
            createdAt: createdAt
        )
    }
}

extension HealthInsightEntity {
    public func toDomain() -> HealthInsight {
        HealthInsight(
            id: id,
            userId: userId,
            type: type,
            title: title,
            description: descriptionText,
            metricValue: metricValue,
            timestamp: timestamp,
            createdAt: createdAt
        )
    }
}
